cluster.go

```diff
@@ -39,38 +41,48 @@ func NewVersionedConfig(version int, data []byte) VersionedConfig {
 	}
 }
 
-// A Backend defines a type of storage backend (m3db, hbase, etc)
-type Backend interface {
+// A Type defines a type of storage (m3db, hbase, etc)
+type Type interface {
 	// Name is the name of the storage class
 	Name() string
-	SetName(s string) Backend
+	SetName(s string) Type
+}
+
+// A Database holds datapoints up to a certain amount of time
+type Database interface {
+	Name() string
+	MaxRetention() time.Duration
 }
 
 // A Cluster defines a cluster
 type Cluster interface {
 	Name() string            // the name of the cluster
-	Backend() Backend        // the cluster's storage class
+	Type() Type              // the cluster's storage type
 	Config() VersionedConfig // the cluster's configuration
+	Database() string        // the database to which the cluster belongs
```

Comment:
I didn't want to force querying for the full Database object here, in case providers kept it in a separate table.  Maybe that doesn't matter though.